src/Utilities/Utilities.jl

```diff
@@ -363,12 +363,31 @@ end
 
 # Finding URLs -- based partially on code from the main Julia repo in `base/methodshow.jl`.
 
```

Comment:
I was thinking that it wouldn't hurt to document this path stuff here since it's not necessarily obvious what's going on in url function. I think it could be more concise and worded better, but I'd propose something along the lines:

```
# Paths on Windows contain backslashes that we need to get rid of in the URLs.
# However, the exact formatting is not consistent and depends on whether Julia runs separately
# or in Cygwin etc.
#
# We get paths from Julia's docsystem, e.g.
#     Docs.docstr(Docs.Binding(Documenter,:Documenter)).data[:path]
# and also from git
#     git rev-parse --show-toplevel
#
# * Ordinary Windows binaries (both Julia and git)
#   In this case the paths from the docsystem are Windows-like with backslashes (e.g. C:\Users\<user>\.julia\v0.6\Documenter\src\Documenter.jl),
#   but the paths from git have forward slashes (e.g. C:/Users/<user>julia/v0.6/Documenter).
#
# * Running under Cygwin, with files outside of Cygwin
#   The paths from the docsystem are still Windows-like with backslashes (e.g. C:\Users\<user>\.julia\v0.6\Documenter\src\Documenter.jl).
#   However, git returns a UNIX-y path using the /cygdrive mount (e.g. /cygdrive/c/<user>/.julia/v0.6/Documenter).
#   We fix that with `cygpath -m <path>` to get a Windows-like path with forward slashes (e.g. C:/Users/<user>/.julia/v0.6/Documenter)
```